### PR TITLE
added the ability for a user to make a new directory

### LINKED ISSUE #1040: New directory here
In the file manager at left, you can right-click a file or folder and choose "New file here". But there's nothing for making new directories. I would like to have a "New directory here" button, similar to "New file here".

### PR DESCRIPTION
**Pull request by TyGuy54**
resolved #1040
This is my first time to contributing on open source projects. Hopefully this code is helpfully!

### CODE REVIEW

File: porcupine/plugins/filemanager.py [hunk 1]
```diff
@@ -149,6 +154,98 @@ def update_dialog_state(*junk: object) -> None:
 
     return new_path
 
+def ask_directory_name(
```
Comment (Akuli) [inline porcupine/plugins/filemanager.py:157]: I don't like the copy/paste. Is there a reason why we can't just use `ask_file_name`?

File: porcupine/plugins/filemanager.py [hunk 2]
```diff
@@ -358,12 +455,22 @@ def new_file_here(path: Path) -> None:
         get_tab_manager().event_generate("<<FileSystemChanged>>")
         get_tab_manager().open_file(name)
 
+def new_directory_here(path: Path) -> None:
+    # print('New Directory Created')
+    name = ask_directory_name(path, "", mode=DirectoryName.NEW)
+    if path.is_dir():
+        path=path.parent
+        name.mkdir()
```
Comment (Akuli) [inline porcupine/plugins/filemanager.py:463]: If I right-click a directory, choose "New directory here", and then click Cancel, I get:
```
Traceback (most recent call last):
  File "/usr/lib/python3.9/tkinter/__init__.py", line 1892, in __call__
    return self.func(*args)
  File "/home/akuli/porcu/porcupine/plugins/filemanager.py", line 463, in new_directory_here
    name.mkdir()
AttributeError: 'NoneType' object has no attribute 'mkdir'
```

File: porcupine/plugins/filemanager.py [hunk 3]
```diff
@@ -32,6 +32,11 @@ class FilenameMode(enum.Enum):
     NEW = ("New file", "Enter a name for the new file. Remember the extension (e.g. .py).")
     PASTE = ("File conflict", "There is already a file named {name} in {parent}.\n\n{plus}")
 
+class DirectoryName(enum.Enum):
+    RENAME = ("Rename", "Enter a new name for {name}:")
+    NEW = ("New directory", "Enter a name for the new directory.")
+    PASTE = ("Directory conflict", "There is already a Directory named {name} in {parent}.\n\n{plus}")
```
Comment (Akuli) [inline porcupine/plugins/filemanager.py:38]: These could probably be just inside `FilenameMode`. Something like this:

```python3
class FilenameMode:
    RENAME = (...)
    NEW_FILE = (...)
    NEW_DIRECTORY = (...)
    PASTE_FILE = (...)
    PASTE_DIRECTORY = (...)
```

**Review by Akuli — COMMENTED** [review]
Thanks!

**Comment by TyGuy54** [comment]
All the above issues should be fixed. 
I'm not sure what I was thinking with the whole copying and pasting, It kept me up last night haha! But for the error you got when you hit cancel when making a new directory was caused by where the `name.mkdir()` it needed to be in the `if name:` block in the `new_directory_here ` function.

File: porcupine/plugins/filemanager.py [hunk 4]
```diff
@@ -150,6 +150,35 @@ def update_dialog_state(*junk: object) -> None:
...
+    entry.bind("<Escape>", (lambda event: cancel_button.invoke()), add=True)
+    entry.select_range(0, "end")
+    entry.focus()
+
+    dialog.title(dialog_title)
+    dialog.resizable(False, False)
+    dialog.wait_window()
+
+    return new_path
+
```
Comment (Akuli) [inline porcupine/plugins/filemanager.py:181]: You apparently forgot to delete some of the copy/paste. There's still two `return new_path`s, and the code after the first one never runs.

File: porcupine/plugins/filemanager.py [hunk 5]
```diff
@@ -352,18 +381,29 @@ def new_file_here(path: Path) -> None:
...
         get_tab_manager().event_generate("<<FileSystemChanged>>")
-        get_tab_manager().open_file(name)
 
 
 commands = [
     # Doing something to an entire project is more difficult than you would think.
     # For example, if the project is renamed, venv locations don't update.
     # TODO: update venv locations when the venv is renamed
     Command("New file here", "<<FileManager:New file>>", (lambda p: True), new_file_here),
+    Command("New directory here", "<<FileManager:New directory>>", (lambda p: True), new_directory_here),
```
Comment (Akuli) [inline porcupine/plugins/filemanager.py:406]: If you right-click a file, there should be no "New directory here" option.
```suggestion
    Command("New directory here", "<<FileManager:New directory>>", (lambda p: p.is_dir()), new_directory_here), 
```

Comment (Akuli) [reply porcupine/plugins/filemanager.py:406]: Nevermind. I forgot about the whole `path=path.parent` thing.

**Comment by Akuli** [comment]
You need to run `black porcupine/plugins/filemanager.py` (see the newly updated README), and then this PR is done :)

**Review by Akuli — APPROVED** [review]
Thanks!

**Pull request merged**